gotham/src/extractor/internal.rs

```diff
@@ -765,8 +766,8 @@ mod tests {
         assert_eq!(p.u16_val, 40511);
         assert_eq!(p.u32_val, 4000000000);
         assert_eq!(p.u64_val, 9000000000);
-        assert_eq!(p.f32_val, 1.4);
-        assert_eq!(p.f64_val, 2.6);
+        assert!((p.f32_val - 1.4).abs() < std::f32::EPSILON);
+        assert!((p.f64_val - 2.6).abs() < std::f64::EPSILON);
```

Comment:
This looks bizarre to me; what exactly is Clippy complaining about? This is a totally non-obvious thing to stumble across in the codebase.

Reply:
Clippy complains that performing strict equality check on two floating point numbers obtained using different code paths is a bad idea. Even if you expect mathematically same results, results of actual floating point computations may differ because precision is not infinite.

This particular code worked before because both values was calculated using the same algorithm: conversion from string to f32/f64. First value - by Gotham code, second - by rustc in compile time. Anyway, "1.4" converted to the binary floating point format is not actually equals to the real world 1.4. Easy experiment using python:

```
>>> "%.20f" % 1.4
'1.39999999999999991118'
```

However, if you think that old code is still good enough, there is way to allow specific problems on selected blocks of code using Clippy.

Reply:
It's more that I'm not entirely sure we should care (yet) about test code & Clippy. People don't write tests as it is, don't want to put them off more due to weird things like this rule.